llm-agents/src/steps/url_scraper.py

```diff
@@ -12,19 +12,19 @@
 #  or implied. See the License for the specific language governing
 #  permissions and limitations under the License.
 
-from typing import List
+from typing import Annotated, List
 
 from steps.url_scraping_utils import get_all_pages, get_nested_readme_urls
-from zenml import step
+from zenml import step, log_artifact_metadata
 
 
-@step(enable_cache=True)
+@step(enable_cache=False)
```

Comment:
we should put all config in a YAML file tbh. Btw why do we disable cache here?

Reply:
Sometimes, if the step fails and the URLs are not loaded, the next pipeline run skips the step and goes into the next step with a None object. This leads to some obscure errors. Doesn't happen usually but it's an edge case that the DSD guys ran into, sometime around the first bootcamp.